ui/static/ui/js/utils.jsx

```diff
@@ -116,6 +116,12 @@ define("utils", ["jquery", "lodash", "react", "select2"],
         allowClear = !isMultiTerms;
       }
 
+      // Allow terms on the fly if the vocabulary is defined to do so
+      var allowTags = false;
+      if (this.props.vocabularyType === 'f') {
```

Comment:
Can you change `this.props.vocabularyType` to `this.props.allowTags` and make it a boolean? Ideally the `Select2` component should be reusable in many different contexts so we shouldn't use vocabulary or term-specific wording